MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/camera/SlowlyRotatingCameraActivity.java

```diff
@@ -0,0 +1,151 @@
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.view.animation.LinearInterpolator;
+import android.widget.Toast;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+
+/**
+ * Animate the map's camera to slowly spin around a single point ont the map.
+ */
+public class SlowlyRotatingCameraActivity extends AppCompatActivity implements OnMapReadyCallback,
+  MapboxMap.OnMapClickListener {
+
+  private static final int DESIRED_NUM_OF_SPINS = 5;
+  private static final int DESIRED_SECONDS_PER_ONE_FULL_360_SPIN = 40;
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private ValueAnimator animator;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_camera_slow_spin);
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(final MapboxMap mapboxMap) {
+    mapboxMap.setStyle(Style.SATELLITE_STREETS, new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        SlowlyRotatingCameraActivity.this.mapboxMap = mapboxMap;
+
+        mapboxMap.addOnMapClickListener(SlowlyRotatingCameraActivity.this);
+
+        // Toast instructing user to tap on the map
+        Toast.makeText(
+          SlowlyRotatingCameraActivity.this, getString(R.string.rotating_camera_toast_instruction),
+          Toast.LENGTH_LONG).show();
+
+        startMapCameraSpinningAnimation(mapboxMap.getCameraPosition().target);
+      }
+    });
+  }
+
+  @Override
+  public boolean onMapClick(@NonNull LatLng point) {
+    startMapCameraSpinningAnimation(point);
+    return true;
+  }
+
+  /**
+   * Set up and start the spin animation. The Android system ValueAnimator emits a new value and that value is
+   * used as the map camera's new bearing rotation amount. A smooth "new helicopter" type of effect is created
+   * by using a LinearInterpolator.
+   *
+   * @param mapCameraTargetLocation the map location that the map camera should spin around
+   */
+  private void startMapCameraSpinningAnimation(@NonNull final LatLng mapCameraTargetLocation) {
```

Comment:
we need to check if animator is not null and cancel it before creating a new animator

Reply:
fixed